implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheus/PrometheusMeterRegistry.java

```diff
@@ -572,13 +557,26 @@ private void onMeterRemoved(Meter meter) {
         }
     }
 
+    private MetricMetadata getMetadata(Meter.Id id) {
+        return getMetadata(id, "");
+    }
+
+    private MetricMetadata getMetadata(Meter.Id id, String suffix) {
+        // TODO: Use the Prometheus naming convention instead
```

Comment:
I fixed this. It worth to mention that:
- At this point Prometheus prefers `.` instead of `_` and the `. -> _` conversion happens later. But it _seems_ using `_` is not a problem here.
- The new Prometheus client validates Counter names if they end with `_total`. If they don't, it appends `_total`. If they do, it throws an exception:
```
java.lang.IllegalArgumentException: 'cnt_total': Illegal metric name.
The metric name must not include the '_total' suffix.
Call PrometheusNaming.sanitizeMetricName(name) to avoid this error.
```
Because of this behavior, if we want to use `PrometheusNamingConvention`, we need to modify it so that it does not add `_total`.